taskcluster/translations_taskgraph/transforms/worker_env.py

```diff
@@ -52,3 +61,23 @@ def inject_worker_env(config, jobs):
         job["worker"]["env"].update(worker_env)
 
         yield job
+
+
+@transforms.add
+def inject_secret(config, jobs):
+    for job in jobs:
+        secret_name = job.pop("use-secret", None)
+
+        # Secret access & configuration through proxy
+        if secret_name is not None:
+            job["worker"]["env"].update(
+                {
+                    "TASKCLUSTER_SECRET": secret_name,
+                }
+            )
+            job["worker"]["taskcluster-proxy"] = True
+            if "scopes" not in job:
+                job["scopes"] = []
+            job["scopes"] += [f"secrets:get:{secret_name}"]
+
+        yield job
```

Comment:
suggestion: please put this in the `kind` instead. We've been trying to keep as much as possible in there to improve readability. It does increase repetition, but it makes it easier to map tasks back to kinds and make tweaks. (If you feel this must go in transform, please use a new one -- this one is _specifically_ for dealing with worker-type specific configuration, which this is not.)

Reply:
I moved these settings to kind in 32f5c4e